src/sentry/models/grouphashtombstone.py

```diff
@@ -0,0 +1,87 @@
+from __future__ import absolute_import, print_function
+
+import logging
+
+from django.db import models, transaction, IntegrityError
+from django.utils import timezone
+
+from sentry.models import GroupHash
+from sentry.db.models import FlexibleForeignKey, Model
+
+
+logger = logging.getLogger(__name__)
+
+
+class GroupHashTombstone(Model):
+    __core__ = True
+
+    project = FlexibleForeignKey('sentry.Project')
+    hash = models.CharField(max_length=32)
+    deleted_at = models.DateTimeField(default=timezone.now)
+
+    class Meta:
+        app_label = 'sentry'
+        db_table = 'sentry_grouphashtombstone'
+        unique_together = (('project', 'hash'), )
+
+    @classmethod
+    def tombstone_groups(cls, project_id, group_ids):
```

Comment:
Definitely open to this living somewhere else under a different name. I didn't want to take over an ORM method like `GroupHash.objects.delete` because there are reasons to delete `GroupHash` without adding tombstones (merge/unmerge).

Also, I'm not sure if I can spare any queries here. It feels a little heavy but nothing is obvious to me.